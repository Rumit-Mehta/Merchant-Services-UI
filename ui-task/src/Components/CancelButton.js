import React from "react";
import Button from "@material-ui/core/Button";
import { useTwitterBtnStyles } from "@mui-treasury/styles/button/twitter";
import { usePushingGutterStyles } from "@mui-treasury/styles/gutter/pushing";
import { Alert } from "@material-ui/lab";
import Snackbar from "@material-ui/core/Snackbar";

export default function SimpleSnackbar() {
  const [open, setOpen] = React.useState(false);

  const handleClick = () => {
    setOpen(true);
  };

  const handleClose = (event, reason) => {
    setOpen(false);
  };

  const styles = useTwitterBtnStyles();
  const mainStyles = usePushingGutterStyles({
    cssProp: "marginTop",
    space: 2,
    firstExcluded: true,
  });
  const wrapperStyles = usePushingGutterStyles();

  return (
    <div className={mainStyles.parent}>
      <div className={wrapperStyles.parent}>
        <Button
          classes={styles}
          variant={"outlined"}
          color={"primary"}
          size={"large"}
          onClick={handleClick}
        >
          Cancel
        </Button>
        <Snackbar open={open} autoHideDuration={3000} onClose={handleClose}>
          <Alert onClose={handleClose} severity="info">
            Going back to previous page
          </Alert>
        </Snackbar>
      </div>
    </div>
  );
}
