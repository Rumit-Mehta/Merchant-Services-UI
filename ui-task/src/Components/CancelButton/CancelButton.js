import React from "react";
import Button from "@material-ui/core/Button";
import { Alert } from "@material-ui/lab";
import Snackbar from "@material-ui/core/Snackbar";
import { createMuiTheme, MuiThemeProvider } from "@material-ui/core/styles";

// This const overides the current theme and adds colours to the outline and the text aswell
// as making the button round
const theme = createMuiTheme({
  overrides: {
    MuiButton: {
      outlined: {
        borderRadius: 25,
      },
      outlinedPrimary: {
        borderColor: "#1176b6",
        color: "#1176b6",
      },
    },
  },
});

// Default function that returns a button with onclick and snackbar functionality. In this case
// the snackbar is only returning a default string with the blue (info) colour
export default function SimpleSnackbar(prop) {
  return (
    <MuiThemeProvider theme={theme}>
      <div data-testid="cancelButtonDiv">
        <Button
          id="CancelButton"
          variant="outlined"
          color="primary"
          size={"large"}
          onClick={prop.handleClick}
        >
          Cancel
        </Button>
        <Snackbar
          open={prop.open}
          autoHideDuration={3000}
          onClose={prop.handleClose}
        >
          <Alert onClose={prop.handleClose} severity="info">
            Going back to previous page
          </Alert>
        </Snackbar>
      </div>
    </MuiThemeProvider>
  );
}
