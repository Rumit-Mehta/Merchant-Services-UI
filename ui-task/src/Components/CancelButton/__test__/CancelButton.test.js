import React from "react";
import ReactDOM from "react-dom";
import CancelButton from "../CancelButton";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("Tests for Cancel Button", () => {
  test("renders  without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<CancelButton></CancelButton>, div);
  });

  const mockClick = jest.fn();
  const mockCancel = jest.fn();
  test("matches snapshot", () => {
    const wrapper = shallow(
      <CancelButton handleClick={mockClick} handleClose={mockCancel} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("matches snapshot", () => {
    const wrapper = shallow(
      <CancelButton open handleClick={mockClick} handleClose={mockCancel} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
