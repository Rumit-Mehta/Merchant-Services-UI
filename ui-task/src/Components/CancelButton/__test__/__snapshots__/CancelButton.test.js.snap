// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Cancel Button matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Tests for Cancel Button matches snapshot 2`] = `ShallowWrapper {}`;
